### ISSUE TITLE
Add flag that stops running unit tests on the first fail

### ISSUE THREAD

**Issue by FractalArt** [mesonbuild/meson#4441]
The way meson seems to handle running tests, is to run them in parallel but to print the information on their passing or failure sequentially. Hence, if a first test takes long to run and the second fails immediately, the report of this failure will only be displayed after the first test has finished running.

Especially during development, it might be good to get immediate notification of a failure and not have to wait for everything to finish in case a test fails. Maybe some flag of the form

```
meson test --stop-on-fail
```

would be useful.

Here is a minimal example displaying the mentioned behavior:
[mesontest.tar.gz](https://github.com/mesonbuild/meson/files/2523208/mesontest.tar.gz)

**Comment by pwithnall** [comment]
This would also be particularly useful in combination with the `--repeat` argument, where it would be nice to have a way to get a racy test to repeat (e.g.) 500 times *or until it first fails*

**Comment by Ericson2314** [comment]
`meson test --maxfail` exists

**Comment by eli-schwartz** [comment]
Aha, yes. This got re-reported in #9352 and implemented in #10837
